tests/transaction.rs

```diff
@@ -0,0 +1,178 @@
+
+        roi.append_field(self.fee_payer_pk.x);
+        roi.append_field(self.source_pk.x);
+        roi.append_field(self.receiver_pk.x);
+
+        roi.append_u64(self.fee);
+        roi.append_u64(self.fee_token);
+        roi.append_bit(self.fee_payer_pk.is_odd);
+        roi.append_u32(self.nonce);
+        roi.append_u32(self.valid_until);
+        roi.append_bytes(self.memo.to_vec());
+
+        for tag_bit in self.tag {
+            roi.append_bit(tag_bit);
+        }
+
+        roi.append_bit(self.source_pk.is_odd);
+        roi.append_bit(self.receiver_pk.is_odd);
+        roi.append_u64(self.token_id);
+        roi.append_u64(self.amount);
+        roi.append_bit(self.token_locked);
+
+        roi
+    }
+
+    fn domain_string(self, network_id: NetworkId) -> &'static str {
+        // Domain strings must have length <= 20
+        match network_id {
+            NetworkId::MAINNET => "MinaSignatureMainnet",
+            NetworkId::TESTNET => "CodaSignature",
+        }
+    }
+}
+
+impl Transaction {
+    pub fn new_payment(from: PubKey, to: PubKey, amount: u64, fee: u64, nonce: u32) -> Self {
+        Transaction {
+            fee: fee,
+            fee_token: 1,
+            fee_payer_pk: from.to_compressed(),
+            nonce: nonce,
+            valid_until: u32::MAX,
+            memo: array_init::array_init(|i| (i == 0) as u8),
+            tag: PAYMENT_TX_TAG,
+            source_pk: from.to_compressed(),
+            receiver_pk: to.to_compressed(),
+            token_id: 1,
+            amount: amount,
+            token_locked: false,
+        }
+    }
+
+    pub fn new_delegation(from: PubKey, to: PubKey, fee: u64, nonce: u32) -> Self {
+        Transaction {
+            fee: fee,
+            fee_token: 1,
+            fee_payer_pk: from.to_compressed(),
+            nonce: nonce,
+            valid_until: u32::MAX,
+            memo: array_init::array_init(|i| (i == 0) as u8),
+            tag: DELEGATION_TX_TAG,
+            source_pk: from.to_compressed(),
+            receiver_pk: to.to_compressed(),
+            token_id: 1,
+            amount: 0,
+            token_locked: false,
+        }
+    }
+
+    pub fn set_valid_until(mut self, global_slot: u32) -> Self {
+        self.valid_until = global_slot;
+
+        self
+    }
+
+    pub fn set_memo(mut self, memo: [u8; MEMO_BYTES - 2]) -> Self {
+        self.memo[0] = 0x01;
+        self.memo[1] = (MEMO_BYTES - 2) as u8;
+        self.memo[2..].copy_from_slice(&memo[..]);
+
+        self
+    }
+
+    pub fn set_memo_str(mut self, memo: &str) -> Self {
+        self.memo[0] = 0x01;
+        self.memo[1] = std::cmp::min(memo.len(), MEMO_BYTES - 2) as u8;
+        let memo = format!("{:\0<32}", memo); // Pad user-supplied memo with zeros
+        self.memo[2..].copy_from_slice(&memo.as_bytes()[..std::cmp::min(memo.len(), MEMO_BYTES - 2)]);
+        // Anything beyond MEMO_BYTES is truncated
+
+        self
+    }
+}
+
+
+#[cfg(test)]
```

Comment:
you don't need this when you're in a tests/ folder

Reply:
Fixed!